### DISCUSSION TITLE
How can I preserve existing parameters while upgrading from 4.19 to 4.20, ensuring new parameters are replaced, added, or removed as needed?

### DISCUSSION THREAD

**Discussion by shaerul** [apache/cloudstack#10071] [Q&A] [answered]
When upgrading the management and agent servers from version 4.19 to 4.20 on Ubuntu 20.04, the process pauses at one point, presenting four options. 

10.190.113.11 Configuration file '/etc/cloudstack/management/server.properties'
10.190.113.11  ==> Modified (by you or by a script) since installation.
10.190.113.11  ==> Package distributor has shipped an updated version.
10.190.113.11    What would you like to do about it ?  Your options are:
10.190.113.11     Y or I  : install the package maintainer's version
10.190.113.11     N or O  : keep your currently-installed version
10.190.113.11       D     : show the differences between the versions
10.190.113.11       Z     : start a shell to examine the situation
10.190.113.11  The default action is to keep your current version.
10.190.113.11 *** server.properties (Y/I/N/O/D/Z) [default=N] ?

I chose option "D" and noticed that selecting the default or "keep your currently-installed version" option would exclude many new parameters. Conversely, choosing "Y" would result in the loss of old parameters. 

How can I preserve the existing parameters while also replacing, adding, or removing new parameters and their values?

**Comment by DaanHoogland** [comment, marked as answer]
'Z' would be your option @shaerul . this is the ubuntu way. You should be able to use tools like `diff` and the like on the command line to compare and merge new changes in your file, after which you can exit the shell, and keep the "old" version you just created.
